dali/pipeline/util/worker_thread.h

```diff
@@ -47,10 +47,14 @@ class Barrier {
         }
     }
     void Break() {
-      count_ = 0;
-      current_ = count_;
+      {
+        std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
minor: lock_guard is cheaper when you just need RAII

Reply:
done